### ISSUE TITLE
Why doesn't Button work?

### ISSUE THREAD

**Issue by PySimpleGUI** [Saadmairaj/tkmacosx#2]
I'm really stunned and confused why this problem was reported in 2009 or before.  Here's an StackOverflow post from 2009 that talks about this very problem.
https://stackoverflow.com/questions/1529847/how-to-change-the-foreground-or-background-colour-of-a-tkinter-button-on-mac-os

I've been struggling to get Buttons to work on the Mac for the past year for the PySimpleGUI project.  I'm planning on switching to using ttk.Button for the Mac, but I don't have the styling of ttk objects working right yet.  Until I get that right, adding ttk Buttons won't help very much.

Can you explain why this problem remains after 10 years?  Why it's only on the Mac this problem happens (not being able to change the background color of buttons)?

**Comment by Saadmairaj** [comment]
There can be multiple reasons why we cannot change the background color of the buttons on the mac. Like now python.org comes with its own version of Tcl/Tk version 8.6.8 which is different from Apple  Tcl/Tk - 8.5.9 which makes it hard to communicate with the os. This is also a problem when you want a dark mode enabled window with Tkinter, see this [post](https://stackoverflow.com/a/55516222/10364425) on StackOverflow to get a better understanding. Another reason can be that Apple doesn't want its widgets to change much.

**Comment by PySimpleGUI** [comment]
It's taken 2 years, but I finally got the answers I was after.

Basically tkinter through homebrew is horrifically broken and it's quite clear will not be repaired.  

I ended up implementing ttk buttons as well as tk buttons so Mac users could use the ttk ones.

Also, my users began to download and use Python from python.org and all of their problems magically went away.  Every person that's reported a problem and switched to the python.org version had their problems go away.  I think that even the tk buttons work now which would be ideal for my package PySimpleGUI.

It's been a real eye opener seeing how rough it's been on the Mac community wanting to build GUI apps using Python.  I would have bet any amount of money that Apple would have made sure Python on the Mac worked flawlessly when it came to GUIs, but I had it completely backwards.  It's as if they don't want people to be successful at it.  It's another head-scratching situation.  

The fall Apple has taken over the past decade has been an amazing sight.  I've also learned that while Mac users can be a vocal bunch, they represent such a tiny fraction of the community that the actual number of people impacted is low.  Only about 7% of my users use Mac.  I try my best to help them, but sometimes they're simply out of reach and have too many things actively working against them.

Thanks for the explanation.  I had no idea they were still shipping 8.5 with anything.  It's no wonder there's bugs everywhere.
